server/controllers/finance/reports/client_support/index.js

```diff
@@ -0,0 +1,140 @@
+          (gl.debit_equiv - gl.credit_equiv) AS balance, em.text AS reference
+        FROM general_ledger gl 
+        JOIN account a ON a.id = gl.account_id
+        JOIN creditor c ON c.uuid = gl.entity_uuid
+        JOIN employee e ON e.creditor_uuid = c.uuid
+        JOIN patient p ON p.uuid = e.patient_uuid
+        JOIN entity_map em ON em.uuid = e.creditor_uuid
+        WHERE (gl.trans_date BETWEEN ? AND ?) AND gl.transaction_type_id = ${SUPPORT_TRANSACTION_TYPE}
+      ) z ON z.trans_id = i.trans_id 
+      WHERE (i.trans_date BETWEEN ? AND ?) AND i.transaction_type_id = ${SUPPORT_TRANSACTION_TYPE} 
+    `;
+
+    const otherSupportQuery = `
+      SELECT 
+        a.label, dg.name, p.display_name,
+        SUM(i.credit_equiv - i.debit_equiv) AS balance, z.label AS recipient_account
+      FROM general_ledger i 
+      JOIN account a ON a.id = i.account_id
+      JOIN debtor d ON d.uuid = i.entity_uuid
+      JOIN debtor_group dg ON dg.uuid = d.group_uuid
+      JOIN patient p ON p.debtor_uuid = d.uuid
+      JOIN (
+        SELECT gl.trans_id, a.label, (gl.debit_equiv - gl.credit_equiv) AS balance, gl.entity_uuid
+        FROM general_ledger gl 
+        JOIN account a ON a.id = gl.account_id
+        WHERE (gl.trans_date BETWEEN ? AND ?) 
+          AND gl.transaction_type_id = ${SUPPORT_TRANSACTION_TYPE} AND gl.entity_uuid IS NULL
+      ) z ON z.trans_id = i.trans_id
+      WHERE (i.trans_date BETWEEN ? AND ?) 
+        AND i.transaction_type_id = ${SUPPORT_TRANSACTION_TYPE} 
+    `;
+
+    const groupByDebtor = ' GROUP BY d.uuid ORDER BY p.display_name; ';
+    const parameters = [dateFrom, dateTo, dateFrom, dateTo];
+
+    const employeeSupportTotal = await db.one(employeeSupportQuery, parameters);
+    const employeeSupport = await db.exec(employeeSupportQuery.concat(groupByDebtor), parameters);
+
+    const otherSupportTotal = await db.one(otherSupportQuery, parameters);
+    const otherSupport = await db.exec(otherSupportQuery.concat(groupByDebtor), parameters);
+
+    const employeesCollection = generateTree(employeeSupport, 'employee_name')
+      .map(e => {
+        // organize tree by debtor group name
+        e.data = generateTree(e.data, 'name');
+        return e;
+      });
+
+    const othersCollection = generateTree(otherSupport, 'recipient_account')
+      .map(e => {
+        // organize tree by debtor group name
+        e.data = generateTree(e.data, 'name');
+        return e;
+      });
+
+    const result = await rpt.render({
+      employeeSupportTotal,
+      employeesCollection,
+      otherSupportTotal,
+      othersCollection,
+      dateFrom,
+      dateTo,
+      showDetails,
+      showEmployeeSupport,
+      showOtherSupport,
+    });
+
+    res.set(result.headers).send(result.report);
+  } catch (e) {
+    next(e);
+  }
+}
+
+function generateTree(array, groupBy) {
+  const collection = _(array)
+    .groupBy(groupBy)
+    .reduce((prev, curr, key) => {
```

Comment:
It looks like a `map()` is better suited here.  You are taking in an array and producing an array, right?